From 6b9c7c6a2f7833a4eb783eaeac57b000f88de30f Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 11 May 2020 13:34:40 +0100
Subject: [PATCH] mh: fix creating new mailbox

Fixes: #2324
---
 maildir/mh.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/maildir/mh.c b/maildir/mh.c
index 09bafbf7aa..0b95061377 100644
--- a/maildir/mh.c
+++ b/maildir/mh.c
@@ -559,12 +559,10 @@ static int mh_mbox_open_append(struct Mailbox *m, OpenMailboxFlags flags)
   if (!m)
     return -1;
 
-  if (!(flags & MUTT_APPENDNEW))
-  {
+  if (!(flags & (MUTT_APPENDNEW | MUTT_NEWFOLDER)))
     return 0;
-  }
 
-  if (mkdir(mailbox_path(m), S_IRWXU))
+  if (mutt_file_mkdir(mailbox_path(m), S_IRWXU))
   {
     mutt_perror(mailbox_path(m));
     return -1;
